modules/core/output_modules.php

```diff
@@ -544,6 +543,15 @@ class Hm_Output_js_data extends Hm_Output_Module {
     protected function output() {
         $res = '<script type="text/javascript">'.
             'var globals = {};'.
+            'window.current_lang = ' . json_encode($this->lang) . ';' .
+            'window.translations = ' . json_encode($this->all_trans()) . ';' .
```

Comment:
No need to pass all the translations from all the languages here. Just pass the current language translations to the JS context.

Reply:
Copy that @kroky , Considering users can write emails in any language, even different from the default, and our need to detect attachments in testing across languages, how would you suggest optimizing the process without losing this flexibility?

Reply:
Yes, that's a good point but will probably make the check quite slower if we keep it like that. Hmm, I don't see another option, however. OK, let's keep it with all languages for now.